cmd/bucket-metadata-sys.go

```diff
@@ -448,14 +488,24 @@ func (sys *BucketMetadataSys) refreshBucketsMetadataLoop(ctx context.Context) {
 				logger.LogIf(ctx, err)
 				continue
 			}
-			for i := range buckets {
-				err := sys.loadBucketMetadata(ctx, buckets[i])
+
+			// Handle if we have some buckets in-memory those are stale.
+			// first delete them and then replace the newer state()
+			// from disk.
+			bucketSet := set.CreateStringSet()
+			for _, bucket := range buckets {
+				bucketSet.Add(bucket.Name)
+			}
+			sys.RemoveStaleBuckets(bucketSet)
```

Comment:
I am not sure I understand what this code does ? why this is called a stale bucket ? we just found this bucket in ListBuckets() call.

Reply:
RemoveStaleBuckets() takes the input of what are the on disk buckets, to remove what is present in memory @vadmeste

Reply:
There was a logical error in RemoveStaleBuckets() its fixed now @vadmeste